common/thorhelper/thorsoapcall.cpp

```diff
@@ -1919,6 +1919,7 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
                 StringBuffer s;
                 master->logctx.CTXLOG("Exception %s - retrying? (%d<%d)", e->errorMessage(s).str(), attempts, master->maxRetries);
 
+                attempts++;
```

Comment:
I think the cause of the confusion is/was that the variable is called (and was tracking) "attempts" but the configuration variable it is testing against is "retries" 

While after the change it will behave correctly I think it would be less confusing if you renamed the variable to "retries", left the increment where it was, and changed the comparison to a >=